gtsam/base/std_optional_serialization.h

```diff
@@ -107,5 +107,6 @@ void serialize(Archive& ar, std::optional<T>& t, const unsigned int version) {
 
 }  // namespace serialization
 }  // namespace boost
-#endif
-#endif
+#endif // BOOST_VERSION < 108400
+#endif // BOOST_VERSION < 108000
```

Comment:
@stevenlovegrove can you please kill this endif and its corresponding #ifdef?

Reply:
> @stevenlovegrove can you please kill this endif and its corresponding #ifdef?

There are two endif's there, and they are guarding different code blocks. Which do you want to kill? Both? I am not familiar enough with Boost to know how widely used these versions are to know the impact of removing the guards.

Reply:
The issue being fixed requires the 108400 block and not the 108000 block, hence we can kill the 108000 block without any impact.